community/kernel/src/main/java/org/neo4j/kernel/api/schema/NodeMultiPropertyDescriptor.java

```diff
@@ -0,0 +1,140 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.schema;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.neo4j.kernel.api.TokenNameLookup;
+
+import static java.lang.String.format;
+
+/**
+ * Description of a single label combined with multiple properties.
+ */
+public class NodeMultiPropertyDescriptor extends NodePropertyDescriptor
+{
+    private final int[] propertyKeyIds;
+
+    public NodeMultiPropertyDescriptor( int labelId, int[] propertyKeyIds )
+    {
+        super( labelId, -1 );
+        this.propertyKeyIds = propertyKeyIds;
+    }
+
+    @Override
+    public boolean equals( Object obj )
+    {
+        if ( this == obj )
+        {
+            return true;
+        }
+        if ( obj != null && obj instanceof NodeMultiPropertyDescriptor )
+        {
+            NodeMultiPropertyDescriptor that = (NodeMultiPropertyDescriptor) obj;
+            return this.getLabelId() == that.getLabelId() &&
+                   Arrays.equals( this.propertyKeyIds, that.propertyKeyIds );
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isComposite()
+    {
+        return true;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = getLabelId();
+        for ( int element : propertyKeyIds )
+        {
+            result = 31 * result + element;
+        }
+        return result;
+    }
+
+    /**
+     * @return property key token id this index is for.
+     */
+    public int getPropertyKeyId()
+    {
+        throw new UnsupportedOperationException( "Cannot get single property Id of composite index" );
+    }
+
+    /**
+     * @return property key token ids this index is for.
+     */
+    public int[] getPropertyKeyIds()
+    {
+        return propertyKeyIds;
+    }
+
+    @Override
+    public String propertyIdText()
+    {
+        return Arrays.stream( propertyKeyIds ).mapToObj( id -> Integer.toString( id ) )
+                .collect( Collectors.joining( "," ) );
+    }
+
+    public String propertyNameText( TokenNameLookup tokenNameLookup )
+    {
+        return Arrays.stream( propertyKeyIds ).mapToObj( id -> tokenNameLookup.propertyKeyGetName( id ) )
+                .collect( Collectors.joining( "," ) );
+    }
+
+    /**
+     * @param tokenNameLookup used for looking up names for token ids.
+     * @return a user friendly description of what this index indexes.
+     */
+    public String userDescription( TokenNameLookup tokenNameLookup )
+    {
+        return format( ":%s(%s)", tokenNameLookup.labelGetName( getLabelId() ),
+                propertyNameText( tokenNameLookup ) );
+    }
+
+    @Override
+    public int compareTo( EntityPropertyDescriptor other )
```

Comment:
The compareTo contract allows any negative and positive numbers. https://docs.oracle.com/javase/7/docs/api/java/lang/Comparable.html#compareTo(T)

This code is based on previously existing code for IndexDescriptor which also did not require -1m 0 or 1.